### PR TITLE
Secure API calls

### PR DESCRIPTION
**Pull request by sergous**
https://github.com/dtpstat/website/issues/50

\* Require auth0 authentication efore use API calls
\* Remove unused API 'users'
\* Restrict user API calls for use only by the owner
\* Handle API incorrect calls

### CODE REVIEW

File: src/pages/api/comments/[accident-id].ts [hunk 1]
```diff
@@ -7,14 +8,30 @@ import { NewComment } from "../../../types";
...
 
-  if (req.method === "POST" && accidentId) {
+  if (!accidentId) {
+    res.status(422).json({
+      error: "missing_param",
+      description: "query param accident-id is missing",
+    });
+
+    return;
+  }
```
Comment (kachkaev) [inline src/pages/api/comments/[accident-id].ts:18]: Not sure this param can be missing. The route is `src/pages/api/comments/[accident-id].ts` and Next should not call the handler if we hit `/api/comments//` etc.

File: src/pages/api/users/[user-id].ts [hunk 2]
```diff
@@ -1,21 +1,54 @@
...
+
+  if (!userId) {
+    res.status(422).json({
+      error: "missing_param",
+      description: "query param user-id is missing",
+    });
+
+    return;
+  }
+
```
Comment (kachkaev) [inline src/pages/api/users/[user-id].ts:28]: Not sure this param can be missing. The route is `src/pages/api/users/[user-id].ts` and Next should not call the handler if we hit `/api/users//` etc.

**Review by kachkaev — COMMENTED** [review]
I haven’t checked the code locally, but the diff looks good! Let’s remove two unnecessary checks before merging (we don’t want to maintain more lines of code than necessary).

It‘d be also nice to change PR title to something like `Do xyz`. This will align squash commit with other commits and `git blame` will look more coherent.

File: src/pages/api/users/[user-id].ts [hunk 3]
```diff
@@ -1,21 +1,54 @@
...
+  if (userId !== sessionUser.sub) {
+    res.status(403).json({
+      error: "forbidden",
+      description: `you are not authorized for access user-id '${userId}'`,
```
Comment (kachkaev) [inline src/pages/api/users/[user-id].ts:32]:
```suggestion
      description: `you are not authorized to access user-id '${userId}'`,
```

```suggestion
      description: `you are not authorized for accessing user-id '${userId}'`,
```

File: src/pages/api/comments/[accident-id].ts (hunk 1 above)
Comment (sergous) [reply src/pages/api/comments/[accident-id].ts:18]: Done

**Review by kachkaev — APPROVED** [review]

**Pull request merged**